src/AutoRest.CSharp/Common/Output/Builders/BuilderHelpers.cs

```diff
@@ -62,9 +62,13 @@ public static Constant ParseConstant(object? value, CSharpType type)
             DateTimeSchema {Format: DateTimeSchemaFormat.DateTime} => SerializationFormat.DateTime_ISO8601,
             DateTimeSchema {Format: DateTimeSchemaFormat.DateTimeRfc1123} => SerializationFormat.DateTime_RFC1123,
             DateSchema _ => SerializationFormat.Date_ISO8601,
-            DurationSchema _ => schema.Extensions?.Format?.Equals("duration-constant") == true ? SerializationFormat.Duration_Constant : SerializationFormat.Duration_ISO8601,
+            DurationSchema _ => schema.Extensions?.Format?.Equals(XMsFormat.DurationConstant) == true ? SerializationFormat.Duration_Constant : SerializationFormat.Duration_ISO8601,
             TimeSchema _ => SerializationFormat.Time_ISO8601,
-            _ => SerializationFormat.Default
+            _ => schema.Extensions?.Format switch
+                {
+                    XMsFormat.DurationConstant => SerializationFormat.Duration_Constant,
```

Comment:
Why we have this again since we already checked this in the section of duration schema? Is it by design that this `x-ms-format: duration-constant` will only take effect together with `format: duration`? Or the `format duration` is actually optional?

Reply:
modelerfour will only convert the schema to duration type if we have `format: duration`. If we only add `format: duration-constant`, modelerfour will treat it as string. By adding this check here and also in TypeFactory to return TimeSpan when we have `format: duration-constant` for a string schema, we no longer require `format: duration` when we already set `format: duration-constant`, which is simpler for users.